src/Microsoft.AspNetCore.Routing/Patterns/RoutePattern.cs

```diff
@@ -21,10 +21,10 @@ public sealed class RoutePattern
 
         internal RoutePattern(
             string rawText,
-            Dictionary<string, object> defaults,
-            Dictionary<string, IReadOnlyList<RoutePatternConstraintReference>> constraints,
-            RoutePatternParameterPart[] parameters,
-            RoutePatternPathSegment[] pathSegments)
+            IReadOnlyDictionary<string, object> defaults,
```

Comment:
This probably makes the performance worse.

Reply:
The property it is assigned to is `IReadOnlyDictionary<string, object>` so it makes no difference.

Reply:
Ah, it was already bad. Maybe make all Dictionary

Reply:
It's a read only dictionary because RoutePattern and friends are immutable.

The interface overhead doesn't effect routing performance because everything is loaded into runtime optimized data structures (DFA matcher state machine).

Reply:
So this overall optimization isn't that impactful is what you're saying?

Reply:
It's cutting allocations at startup.